### ISSUE TITLE
Contribution upstream?

### ISSUE THREAD

**Issue by remcoder** [LiaTemplates/gcode-preview#1]
Hi, I'm the creator of the GCode Preview lib you're using in this repo and I just happened to come across it! Really nice work on wrapping the lib in a WebComponent :-)

I think it is a great idea to wrap the lib in a WebComponent. This is something that could be a valuable addition to the lib actually so I was wondering if you'd be open to contribute your code to the lib?

have a nice day!

**Comment by andre-dietrich** [comment]
@remcoder ... Sure, what would you think would be the best way, a special repository that uses it or as a subproject?

My intention was to use as a library with LiaScript. Therefor, the raw-MD-file only has to be attached as a URL-parameter to:

https://liascript.github.io/course/?https://raw.githubusercontent.com/LiaTemplates/gcode-preview/refs/heads/main/README.md

and your examples become interactive with executable code-blocks ...

**Comment by remcoder** [comment]
My intention would be to offer a Web Component wrapper as part of the gcode-preview npm package, if possible based on your code.

A separate repo could also work, and for me and others to use it, it would need to be on npmjs too. 

Was your intention to open source your code btw? Since I didn't see license, I didn't want to make any assumptions :-) I guess it having some kind of open source license would be the first step in allowing this code to be used in a wider context (if that's what you want).

**Comment by andre-dietrich** [comment]
Would the "no license" fit ... I did not check your license 🙈  ...

**Comment by remcoder** [comment]
I depends on what you want. My lib is released under (checks notes) the MIT license. It means everyone can use the code freely without limitations. Any code that becomes part of my repo will fall automatically under that license. 

If you're ok with that I'll make a branch in my repo with your code and I'll build an example page around it. 

See this ticket from last year https://github.com/xyz-tools/gcode-preview/issues/178

**Comment by andre-dietrich** [comment]
Of course, feel free to use it .... I would be happy to contribute ... not all of the settings are working, but I think with your help this could be easily solved ....

**Comment by remcoder** [comment]
Very cool! I'll let you know when I have an update!

**Comment by remcoder** [comment]
btw feel free to join [our (small) discord](https://discord.gg/QyeefnnX) to chat about development and ideas! :D
